TCMSApp/src/client/app/defects/add-defect.controller.spec.js

```diff
@@ -0,0 +1,125 @@
+/* jshint -W117, -W030 */
+describe('Tests Add-Defects', function() {
+    //this.timeout(15000);
+    var controller, controllerModal, scope, state, useInfo;
+    var Trello = {
+        authorized: function() {
+            return true;
+        },
+        authorize: function() {},
+        get: function() {}
+    };
+    beforeEach(module('app.core'));
+    beforeEach(function() {
+        module(function($provide) {
+            $provide.constant('Trello', Trello);
+        });
+    });
+    beforeEach(function () {
+        bard.appModule('app.defects');
+        bard.inject('$controller', '$rootScope', '$httpBackend', '$state', 'apiUrl', 'user');
+    });
+
+    beforeEach(function () {
+        scope = $rootScope.$new();
+        state = {};
+        state.$current = {};
+        state.$current.name = 'defects';
+        state.go = function(string)
+        {
+            return string;
+        };
+        useInfo = {};
+        useInfo.authorized = function() {
+            return true;
+        };
+        controller = $controller('AddDefectController', {$scope: scope, $state: state, user:useInfo});
+        var httpBackend = $httpBackend;
+        httpBackend.when('GET', apiUrl.user).respond(
+            [
+                {
+                    '_id':'56b323080d43df5011807f2a',
+                    'lastName':'Mart',
+                    'firstName':'Alex',
+                    'email':'tester1@soft.com',
+                    'password':'$2a$10$X1mcILfK.HfqBk3oUsszOeAi2RitZcaVCd4Awl4pmsuBL2vCf0Bti',
+                    '__v':0,
+                    'projects':[]}
+            ]
+        );
+        httpBackend.flush();
+        $rootScope.$digest();
+    });
+
+    bard.verifyNoOutstandingHttpRequests();
+
+    describe('Defects Controller', function () {
+        it('should be created successfully', function () {
+            expect(controller).to.be.defined;
+        });
+    });
+});
+
+describe('Tests Add-Defects2', function() {
```

Comment:
Actually it is an example of code duplication :) 
Tests are almost the same.
Why do we need the second block? How it differs from the first one? Why do we pass `uibModal` and `$stateParams:state`?